x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authc/RealmConfig.java

```diff
@@ -24,11 +24,24 @@
     private final ThreadContext threadContext;
 
     public RealmConfig(RealmIdentifier identifier, Settings settings, Environment env, ThreadContext threadContext) {
+        this(identifier, settings, env, threadContext, null);
+    }
+
+    public RealmConfig(RealmIdentifier identifier, Settings settings, Environment env, ThreadContext threadContext, Integer order) {
         this.identifier = identifier;
         this.settings = settings;
         this.env = env;
         this.enabled = getSetting(RealmSettings.ENABLED_SETTING);
-        this.order = getSetting(RealmSettings.ORDER_SETTING);
+        if (order != null) {
+            this.order = order;
+        } else if (order == null && hasSetting(RealmSettings.ORDER_SETTING.apply(type())) == false) {
+            throw new IllegalArgumentException("'order' is a mandatory parameter for realm config. " +
+                "Found invalid realm config: '" + identifier.name + "'\n" +
+                "Please see the breaking changes documentation."
```

Comment:
Do we have a test for this validation (e.g. `RealmSettingsTests`) ?
I can't see one, but given the number of tests changes, I may have missed it.

Reply:
I haven't added the tests yet mainly because I was not sure how the change will turn out (e.g. the above discussion may result in implementation change). I'll add them once I am certain about the changes.

Reply:
Tests added.